### PR TITLE
feat(firewall): wire up firewaller to use ports domain

### PR DESCRIPTION
**Pull request by jack-w-shaw**
Wire up the firewaller api on the controller facade to be backed by
the DQLite ports domain.
 
This included substituting the DQLite watcher for the Mongo watcher.

This required a quick patch to the DQLite watcher to emit machine names
instead of machine uuids, since this data is consumed outside the
apiserver, so should be a natural key.

We also needed to wire up the OpenedPortRangesForMachine facade
endpoint. This required a bit of hacking (i.e. hardcoding SubnetCIDRs to
nil) to wire up. This was done since the application endpoints domain is
not yet wired up. Restoring this has been added to the risk register.

## Checklist

- [x] Code style: imports ordered, good names, simple structure, etc
- [x] Comments saying why design decisions were made
- [x] Go unit tests, with comments saying what you're testing

## QA steps

(Note: 4.0 tip is currently experiencing some difficulties with controllers (uniter bouncing). You may need to bootstrap a fw times to get a functioning controller)

Run
```
juju bootstrap aws aws
juju add-model m
juju deploy ubuntu
juju expose ubuntu
juju exec --unit ubuntu/0 open-port 1000/tcp
```

Check debug logs for:
```
machine-0: 11:51:28 INFO juju.provider.ec2 opened ports in security group juju-0a8403c4-10cc-4b80-89d3-77ae26bb4b3d-0: [1000/tcp from 0.0.0.0/0,::/0]
```

verify the port 1000 is open on the machine sec group
image

### CODE REVIEW

File: apiserver/facades/controller/firewaller/firewaller.go [hunk 1]
```diff
@@ -69,7 +75,7 @@ type FirewallerAPI struct {
 	logger                                   corelogger.Logger
 
 	// Fetched on demand and memoized
-	appEndpointBindings map[string]map[string]string
+	// appEndpointBindings map[string]map[string]string
```
Comment (SimonRichardson) [inline apiserver/facades/controller/firewaller/firewaller.go:78]: Delete me or keep me.

File: apiserver/facades/controller/firewaller/firewaller.go [hunk 2]
```diff
@@ -323,20 +333,23 @@ func (f *FirewallerAPI) GetAssignedMachine(ctx context.Context, args params.Enti
 	return result, nil
 }
 
+// NOTE(jack-w-shaw): temporarily dropped pending application endpoint DQLite
+// implementation
```
Comment (SimonRichardson) [inline apiserver/facades/controller/firewaller/firewaller.go:337]: I am much more of a fan of creating a task on the application epic, which links to the 3.6 code, which states that it should be reinstated.

File: domain/port/state/watcher.go [hunk 3]
```diff
@@ -23,11 +24,11 @@ func (st *State) WatchOpenedPortsTable() string {
 // InitialWatchOpenedPortsStatement returns the query to load the initial event
 // for the WatchOpenedPorts watcher
 func (st *State) InitialWatchOpenedPortsStatement() string {
-	return "SELECT uuid FROM machine"
+	return "SELECT name FROM machine"
 }
 
 // GetMachinesForEndpoints returns a slice of machine UUIDs that host the provided endpoints.
```
Comment (SimonRichardson) [inline domain/port/state/watcher.go:30]: The comment is wrong.

File: domain/port/state/watcher.go [hunk 4]
```diff
@@ -23,11 +24,11 @@ func (st *State) WatchOpenedPortsTable() string {
...
-func (st *State) GetMachinesForEndpoints(ctx context.Context, eps []string) ([]string, error) {
+func (st *State) GetMachinesForEndpoints(ctx context.Context, eps []string) ([]coremachine.Name, error) {
```
Comment (SimonRichardson) [inline domain/port/state/watcher.go:31]: I wish this returned `map[string]....` then you can see which machines are associated with which endpoint.

File: domain/port/service/watcher.go [hunk 5]
```diff
@@ -103,17 +105,17 @@ func (s *WatchableService) endpointToMachineMapper(
 		return e.Changed()
 	})
 
-	machineUUIDs, err := s.st.GetMachinesForEndpoints(ctx, endpointUUIDs)
+	machineNames, err := s.st.GetMachinesForEndpoints(ctx, endpointUUIDs)
```
Comment (SimonRichardson) [inline domain/port/service/watcher.go:108]: Rename to `GetMachineNamesByEndpoints`

**Review by SimonRichardson — COMMENTED** [review]
The code comments aren't ideal. I think a better approach is to fully document what to implement in a task and add it to the right epic. I'm not a fan of leaving partially implemented code, as it can drift over time and can be harder to implement.

**Review by manadart — COMMENTED** [review]
Needs a little reorganisation.

File: domain/port/service/watcher.go (hunk 5 above)
Comment (jack-w-shaw) [reply domain/port/service/watcher.go:108]: How about `GetMachineNamesForEndpoints`?

**Pull request closed without merging**